lerobot/common/policies/vqbet/modeling_vqbet.py

```diff
@@ -0,0 +1,2902 @@
+            # TODO(rcadene): make _forward return output dictionary?
+            actions = self.unnormalize_outputs({"action": actions})["action"]
+
+            self._action_queue.extend(actions.transpose(0, 1))
+        return self._action_queue.popleft()
+
+    def forward(self, batch: dict[str, Tensor]) -> dict[str, Tensor]:
+        """Run the batch through the model and compute the loss for training or validation."""
+        batch = self.normalize_inputs(batch)
+        batch = self.normalize_targets(batch)
+        if not self.check_discretized():
+            loss = self.vqbet.discretize(self.config.discretize_step, batch['action'])
+            return {"loss": loss}
+        _, loss = self.vqbet(batch)
+        return {"loss": loss['actor_loss']}
+
+
+class VQBeTModel(nn.Module):
+    def __init__(self, config: VQBeTConfig):
+        super().__init__()
+        self.config = config
+
+        self.rgb_encoder = DiffusionRgbEncoder(config)
+
+        global_cond_dim = (config.output_shapes["action"][0] + self.rgb_encoder.feature_dim)
+        self._policy = GPT(
+            GPTConfig(
+                block_size=self.config.block_size,
+                input_dim=global_cond_dim,
+                output_dim=self.config.output_dim,
+                n_layer=self.config.n_layer,
+                n_head=self.config.n_head,
+                n_embd=self.config.n_embd,
+                dropout=self.config.dropout,
+            )
+        )
+        self._action_head = VQBeTHead(
+            config.output_dim,
+            config.output_shapes["action"][0],
+            hidden_size=config.mlp_hidden_dim,
+            vqvae_groups=config.vqvae_groups,
+            vqvae_n_embed=config.vqvae_n_embed,
+            vqvae_embedding_dim=config.vqvae_embedding_dim,
+            n_action_steps=config.n_action_steps
+        )
+
+    def discretize(self, discretize_step, actions):
+        return self._action_head.discretize(discretize_step, actions)
+
+    # ========= inference  ============
+    def forward(self, batch: dict[str, Tensor]) -> Tensor:
+        # Input validation.
+        assert set(batch).issuperset({"observation.state", "observation.image"})
+        batch_size, n_obs_steps = batch["observation.state"].shape[:2]
+        assert n_obs_steps == self.config.n_obs_steps
+
+        # Extract image feature (first combine batch and sequence dims).
+        img_features = self.rgb_encoder(einops.rearrange(batch["observation.image"], "b n ... -> (b n) ..."))
+        # Separate batch and sequence dims.
+        img_features = einops.rearrange(img_features, "(b n) ... -> b n ...", b=batch_size)
+        # Concatenate state and image features then flatten to (B, global_cond_dim).
+        global_cond = torch.cat([batch["observation.state"], img_features], dim=-1)
+
+        obs = global_cond
+        if 'action' in batch.keys():
+            action = batch["action"]
+        else:
+            action = None
+        if 'goal' in batch.keys():
```

Comment:
It was some remnant for "conditional VQ-BeT", but since we are serving only unconditional one now, we dropped it. But will implement it again after implementing some goal-cond envs.